grails-app/migrations/0.9.x/changelog-2024-07-18-1200-alter-table-location-type-make-location-type-code-non-null.groovy

```diff
@@ -0,0 +1,19 @@
+databaseChangeLog = {
+
+    changeSet(author: "ewaterman", id: "180720241200-0", objectQuotingStrategy: "LEGACY") {
```

Comment:
Convention: Also, start the changeset ID from 1 instead of 0 (180720241200-1). I usually reserve the 0 for when a changeset needs to be inserted before the others. However, the ordering of the IDs doesn't really matter as the changesets should be executed in the order they appear, so do whatever feels ok and make sure the IDs are unique. I just wanted to point out the convention.

> [!TIP] 
> 
> Creating these unique IDs is a pain so way back in the day I created a bash script that generates a changeset ID using the date command. There's probably a better way to do this, but it's worked for me. 

Here's the script in case anyone wants to use it. I'll add this to the dbm/liquibase documentation ticket as well. 
```
#!/bin/bash
OUTPUT=$(date +%s%N | cut -b1-13)
echo $OUTPUT
echo -n $OUTPUT | xclip -selection c
```
Copy to $HOME/bin.
```
[jmiranda@jmiranda-ThinkPad-W540 ~]$ ls -al /home/jmiranda/bin/generateChangeSetId.sh 
-rwxrwxr-x 1 jmiranda jmiranda 96 Apr 11  2013 /home/jmiranda/bin/generateChangeSetId.sh
```
NOTE: Holy $#!% I wrote this script over a decade ago. Let me know if there's a better (standard) way to do this (i.e. one improvement would probably to create a bash alias).

Execute bash script (make sure HOME/bin is in the PATH)
```
[jmiranda@jmiranda-ThinkPad-W540 ~]$ generateChangeSetId.sh 
1721751106382

```